mrbgems/mruby-io/mrblib/io.rb

```diff
@@ -296,6 +301,12 @@ def getc
     end
   end
 
+  def getbyte
+    ret = getc
+    return ret.getbyte 0 if ret
```

Comment:
If it is built with `MRB_UTF8_STRING` added, only the first byte of the character seems to be retrieved.

```console
% echo '🌠🌟💫' > test.txt

% bin/mruby -e 'File.open("test.txt") { |f| p f.size.times.map { f.getbyte } }'
[240, 240, 240, 10, nil, nil, nil, nil, nil, nil, nil, nil, nil]

% ruby30 -e 'File.open("test.txt") { |f| p f.size.times.map { f.getbyte } }'
[240, 159, 140, 160, 240, 159, 140, 159, 240, 159, 146, 171, 10]
```

If performance issues don't concern you, it may be better to use `IO#read`.

Reply:
`getbyte` should work with `MRB_UTF8_STRING`.